tortoise/ninja_tortoise.go

```diff
@@ -384,6 +398,14 @@ func (ni *ninjaTortoise) addPatternVote(p votingPattern, view map[types.BlockID]
 			if err != nil {
 				ni.Panic("could not retrieve layer block ids")
 			}
+
+			//explicitly abstain
+			if ex == ZeroPattern {
+				for _, bl := range blocks {
+					ni.tTally[p][bl] = ni.tTally[p][bl].Add(Abstain)
+				}
+			}
```

Comment:
I think that you need to break the for loop in line 396; otherwise you would also add negative votes for the entire (abstained) layer. Anyway, there is no reason to continue with this 'ex'.

Reply:
not break - continue to the next item